mcap_vendor/CMakeLists.txt

```diff
@@ -0,0 +1,87 @@
+cmake_minimum_required(VERSION 3.5)
+project(mcap_vendor)
+
+## Compile options
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+endif()
+
+## Dependencies
+find_package(ament_cmake REQUIRED)
+
+
+## Define vendor macro
+macro(build_mcap_vendor)
+  include(FetchContent)
+  fetchcontent_declare(mcap
+    GIT_REPOSITORY https://github.com/foxglove/mcap.git
+    GIT_TAG releases/cpp/v0.1.1
```

Comment:
I think using commit hash is better.
https://cmake.org/cmake/help/latest/module/FetchContent.html
```
Where contents are being fetched from a remote location and you do not control that server, it is advisable to use a hash for GIT_TAG rather than a branch or tag name. A commit hash is more secure and helps to confirm that the downloaded contents are what you expected.
```

Reply:
> Where contents are being fetched from a remote location *and you do not control that server*

We do control the `foxglove/mcap` repository.

Reply:
How about `zstd` and `lz4`?